proteuslib/unit_models/reverse_osmosis_1D.py

```diff
@@ -0,0 +1,724 @@
+                "reverse osmosis module. "
+                "Defaulting to finite "
+                "difference method."
+            )
+            self.config.transformation_method = "dae.finite_difference"
+
+        if self.config.transformation_scheme is useDefault:
+            _log.warning(
+                "Discretization scheme was "
+                "not specified for the "
+                "reverse osmosis module."
+                "Defaulting to backward finite "
+                "difference."
+            )
+            self.config.transformation_scheme = "BACKWARD"
+
+
+    def build(self):
+        """
+        Build 1D RO model (pre-DAE transformation).
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        self.solvent_list = Set()
+        self.solute_list = Set()
+        self._process_config()
+
+        # Build 1D Control volume for feed side
+        self.feed_side = ControlVolume1DBlock(default={
+            "dynamic": self.config.dynamic,
+            "has_holdup": self.config.has_holdup,
+            "area_definition": self.config.area_definition,
+            "property_package": self.config.property_package,
+            "property_package_args": self.config.property_package_args,
+            "transformation_method": self.config.transformation_method,
+            "transformation_scheme": self.config.transformation_scheme,
+            "finite_elements": self.config.finite_elements,
+            "collocation_points": self.config.collocation_points
+        })
+
+        feed_side = self.feed_side
+        # Add geometry to feed side
+        feed_side.add_geometry()
+        # Add state blocks to feed side
+        feed_side.add_state_blocks(has_phase_equilibrium=False)
+        # Populate feed side
+        feed_side.add_material_balances(balance_type=self.config.material_balance_type,
+                                        has_mass_transfer=True)
+        feed_side.add_momentum_balances(balance_type=self.config.momentum_balance_type,
+                                        has_pressure_change=self.config.has_pressure_change)
+        # Apply transformation to feed side
+        feed_side.apply_transformation()
+        # Add inlet/outlet ports for feed side
+        self.add_inlet_port(name="feed_inlet", block=feed_side)
+        self.add_outlet_port(name="feed_outlet", block=feed_side)
+        # Make indexed stateblock and separate stateblock for permeate-side and permeate outlet, respectively.
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = False  # these blocks are not inlets
+        self.permeate_side = self.config.property_package.state_block_class(
```

Comment:
I think calling these stateblocks "permeate_side" could be confusing because "permeate_side" is a control volume in the 0D model. We should definitely decide whether we want the permeate side to have control volume without mass and energy balances or not and be consistent across the 1D and 0D model.

Reply:
During my first attempt to implement the 1DRO model, I tried setting the permeate_side as a 1D control volume, but it made the implementation trickier. Furthermore, after some discussion in issue #126, I decided to pivot and set the permeate_side as an indexed stateblock. I think this is something we can revisit in a subsequent PR, once the 1DRO model is fully functional. The same applies for any name changing or consistency with the 0DRO model; let's reserve such changes for when functionality of the 1DRO in place. Then we can have a PR that cleans up discrepancies between the 0D and 1D models where possible.

Reply:
I personally wouldn't be too concerned. If someone is digging in deep enough to find this, then they had better know enough to understand what is happening anyway.